tests/system/dcos_service_marathon_tests.py

```diff
@@ -26,19 +28,25 @@ def test_framework_readiness_time_check():
     """Tests that an app is being in deployment until the readiness check is done."""
 
     fw = apps.fake_framework()
-    readiness_time = 30
+    readiness_time = 15
     fw['readinessChecks'][0]['intervalSeconds'] = readiness_time
 
     client = marathon.create_client()
     deployment_id = client.add_app(fw)
 
-    time.sleep(readiness_time - 10)  # not yet.. still deploying
-    deployment = client.get_deployment(deployment_id)
-    assert deployment['currentActions'][0]['readinessCheckResults'][0]['ready'] is False, \
-        "The application is read"
+    @retrying.retry(wait_fixed=1000, stop_max_attempt_number=16, retry_on_exception=common.ignore_exception)
```

Comment:
do we need to retry 16 times? can't we just wait 16 seconds with wait_fixed=16000 ?

Reply:
If we fix the wait time it's like a sleep. I want to avoid that.